File: 160 Intersection of Two Linked Lists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode GetIntersectionNode(ListNode headA, ListNode headB) {
        if(null == headA || null == headB)
            return null;
            
        if(headA == headB)
            return headA;
            
        var dict = new Dictionary<ListNode, int>();
        
        while(headA != null){
            dict.Add(headA, 1);
            headA = headA.next;
        }
        
        while(headB != null){
            if(dict.ContainsKey(headB))
                return headB;
            headB = headB.next;
        }
        
        return null;
    }
}

File: 160 Intersection of Two Linked Lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(headA, headB) {
    if(!headA || !headB)
        return null;
    
    if(headA == headB)
        return headA;
    
    var map = {};
    
    while(headA != null){
        map[headA.val] = headA;
        headA = headA.next;
    }
    
    while(headB != null){
        if(headB.val in map)
            return headB;
        headB = headB.next;
    }
    
    return null;
};